sound/soc/sof/ipc3-topology.c

```diff
@@ -2307,12 +2308,14 @@ static int sof_ipc3_tear_down_all_pipelines(struct snd_sof_dev *sdev, bool verif
 	 * topology loading the sound card unavailable to open PCMs.
 	 */
 	list_for_each_entry(swidget, &sdev->widget_list, list) {
-		if (swidget->dynamic_pipeline_widget)
+		if (swidget->dynamic_pipeline_widget) {
+			dyn_widgets = true;
 			continue;
+		}
 
-		/* Do not free widgets for static pipelines with FW ABI older than 3.19 */
+		/* Do not free widgets for static pipelines with FW older than SOF2.2 */
 		if (!verify && !swidget->dynamic_pipeline_widget &&
```

Comment:
this is redundant (but don't need to be fixed here!):
`swidget->dynamic_pipeline_widget` can only be false as we have skipped the dynamic ones above.

Reply:
Good catch @ujfalusi . I'll resist the urge the address this in the same patch. I'll instead merge this and push a new one for that.